x-pack/plugin/async-search/src/internalClusterTest/java/org/elasticsearch/xpack/search/CrossClusterAsyncSearchIT.java

```diff
@@ -537,11 +660,19 @@ public void testClusterDetailsAfterCCSWithFailuresOnOneClusterOnly() throws Exce
                 ? SearchResponse.Cluster.Status.SKIPPED
                 : SearchResponse.Cluster.Status.FAILED;
             assertThat(remoteClusterSearchInfo.getStatus(), equalTo(expectedStatus));
-            assertNull(remoteClusterSearchInfo.getTotalShards());
-            assertNull(remoteClusterSearchInfo.getSuccessfulShards());
-            assertNull(remoteClusterSearchInfo.getSkippedShards());
-            assertNull(remoteClusterSearchInfo.getFailedShards());
-            assertThat(remoteClusterSearchInfo.getFailures().size(), equalTo(1));
+            if (clusters.isCcsMinimizeRoundtrips()) {
```

Comment:
We can extract something along the lines of:

```java
private static void assertAllShardsFailed(
        SearchResponse.Clusters clusters,
        SearchResponse.Cluster remoteClusterSearchInfo,
        int remoteNumShards
    ) {
        if (clusters.isCcsMinimizeRoundtrips()) {
            assertNull(remoteClusterSearchInfo.getTotalShards());
            assertNull(remoteClusterSearchInfo.getSuccessfulShards());
            assertNull(remoteClusterSearchInfo.getSkippedShards());
            assertNull(remoteClusterSearchInfo.getFailedShards());
            assertThat(remoteClusterSearchInfo.getFailures().size(), equalTo(1));
        } else {
            assertThat(remoteClusterSearchInfo.getTotalShards(), equalTo(remoteNumShards));
            assertThat(remoteClusterSearchInfo.getSuccessfulShards(), equalTo(0));
            assertThat(remoteClusterSearchInfo.getSkippedShards(), equalTo(0));
            assertThat(remoteClusterSearchInfo.getFailedShards(), equalTo(remoteNumShards));
            assertThat(remoteClusterSearchInfo.getFailures().size(), equalTo(remoteNumShards));
        }
        assertNull(remoteClusterSearchInfo.getTook());
        assertFalse(remoteClusterSearchInfo.isTimedOut());
        ShardSearchFailure remoteShardSearchFailure = remoteClusterSearchInfo.getFailures().get(0);
        assertTrue("should have 'index corrupted' in reason", remoteShardSearchFailure.reason().contains("index corrupted"));
    }
```

here as this is duplicated 7x in this file :)

Reply:
Fixed.

Reply:
It's not is it?